### PR TITLE
Added contact/category/tag/page filter to teaser-selection and article-selection

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | none
| License | MIT

#### What's in this PR?

This PR add some additional filter to teaser-selection.

### TODO?

- [x] ok button (Use - Anwenden)
- [x] cancel button (back)
- [x] height
- [x] article-selection

### CODE REVIEW

File: Resources/public/js/components/articles/list/category-selection/form/main.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                        paginationOptions: {
+                            dropdown: {
+                                limit: 20
+                            }
+                        },
+                        childrenPropertyName: 'hasChildren',
+                        viewOptions: {
+                            table: {
+                                cropContents: false,
+                                noItemsText: 'sulu.category.no-categories-available',
```
Comment (trickreich) [inline Resources/public/js/components/articles/list/category-selection/form/main.js]: maybe use translations var here?

File: Resources/public/js/components/articles/list/page-selection/form/main.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                    name: 'page-tree-route/content-datasource@suluarticle',
+                    options: {
+                        el: $container,
+                        selected: this.id,
+                        locale: this.options.locale,
+                        selectedUrl: '/admin/api/nodes/{datasource}?tree=true&language={locale}&fields=title,order,published,url&webspace-nodes=all',
+                        rootUrl: '/admin/api/nodes?language={locale}&fields=title,order,published&webspace-nodes=all',
+                        resultKey: 'nodes',
+                        instanceName: 'internal-link',
+                        instanceNamePrefix: '',
```
Comment (trickreich) [inline Resources/public/js/components/articles/list/page-selection/form/main.js]: is this really needed?

File: Resources/public/js/services/overlay-filter-helper.js [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+        this.$el = $el;
+        this.instanceName = instanceName;
+        this.locale = locale;
+        this.okClickNamespace = okClickNamespace;
+    }
+
+    /**
+     * Start filter components.
+     *
+     * @param {{start}} sandbox
```
Comment (trickreich) [inline Resources/public/js/services/overlay-filter-helper.js]: {{start}}?

File: Resources/public/js/services/overlay-filter-helper.js [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+        getUrlParameter = function(filter) {
+            return {
+                contactId: filter.contact ? filter.contact.id : null,
+                categoryId: filter.category ? filter.category.id : null,
+                tagId: filter.tag ? filter.tag.id : null,
+                pageId: filter.page ? filter.page.id : null,
+            };
+        };
+
+    function FilterState($el, instanceName, locale, okClickNamespace) {
```
Comment (trickreich) [inline Resources/public/js/services/overlay-filter-helper.js]: jsdoc

File: Resources/public/js/services/overlay-filter-helper.js [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+
+    return {
+        /**
+         * Create new filter-state instance.
+         *
+         * @param {Node} $el
+         * @param {String} instanceName
+         * @param {String} locale
+         * @param {String} okClickNamespace
+         * @returns {FilterState}
```
Comment (trickreich) [inline Resources/public/js/services/overlay-filter-helper.js]: empty line

File: Teaser/ArticleTeaserProvider.php [hunk 6]
```diff
@@ -72,11 +72,80 @@ public function getConfiguration()
...
+                            'type' => 'cancel',
+                            'align' => 'left',
+                        ],
+                    ],
+                ],
+                [
+                    'title' => $this->translator->trans('sulu_article.contact-selection-overlay.title', [], 'backend'),
+                    'cssClass' => 'contact-slide',
+                    'contentSpacing' => true,
+                    'okDefaultText' => $this->translator->trans('sulu-content.teaser.apply', [], 'backend'),
```
Comment (trickreich) [inline Teaser/ArticleTeaserProvider.php]: Maybe save this in a var? You use this a few times in here.

File: Resources/public/js/services/overlay-filter-helper.js (hunk 3 above)
Comment (wachterjohannes) [inline Resources/public/js/services/overlay-filter-helper.js]: object with `start` property

File: Resources/public/js/components/articles/list/page-selection/form/main.js (hunk 2 above)
Comment (wachterjohannes) [inline Resources/public/js/components/articles/list/page-selection/form/main.js]: yes :) else `smart-content-´ (default value) will be used in the event-names which is wrong here.

**Comment by wachterjohannes** [comment]
@trickreich fixed or commented

**Pull request merged**